scout/server/blueprints/institutes/controllers.py

```diff
@@ -510,28 +499,53 @@ def populate_case_obj(case_obj):
         case_obj["display_track"] = TRACKS[case_obj.get("track", "rare")]
         return case_obj
 
-    for nr_cases, case_obj in enumerate(all_cases.limit(limit), 1):
-        case_obj = populate_case_obj(case_obj)
-        case_groups[case_obj["status"]].append(case_obj)
+    case_groups = {status: [] for status in CASE_STATUSES}
 
-    if prioritized_cases:
-        extra_prioritized = 0
-        for case_obj in prioritized_cases:
-            if any(
-                group_obj.get("display_name") == case_obj.get("display_name")
-                for group_obj in case_groups[case_obj["status"]]
-            ):
-                continue
-            else:
-                extra_prioritized += 1
-                case_obj = populate_case_obj(case_obj)
-                case_groups[case_obj["status"]].append(case_obj)
-        # extra prioritized cases are potentially shown in addition to the case query limit
-        nr_cases += extra_prioritized
+    nr_cases_showall_statuses = (
+        0  # Nr of cases for the case statuses where all cases should be shown
+    )
+    # In institute settings, retrieve all case status categories for which all cases should be displayed
+    status_show_all_cases: List[str] = institute_obj.get(
+        "show_all_cases_categories", ["prioritized"]
+    )
+    for status in status_show_all_cases:
+        cases_in_status = store.cases_by_status(
+            institute_id=institute_id, status=status, projection=ALL_CASES_PROJECTION
+        )
+        cases_in_status = _sort_cases(data, request, cases_in_status)
+        for case_obj in cases_in_status:
+            populate_case_obj(case_obj)
+            nr_cases_showall_statuses += 1
+            case_groups[status].append(case_obj)
+
+    # Retrieve cases for the remaining status categories
+    all_cases = store.cases(
+        collaborator=institute_id,
+        name_query=name_query,
+        skip_assigned=request.args.get("skip_assigned"),
+        is_research=request.args.get("is_research"),
+        has_rna_data=request.args.get("has_rna"),
+        verification_pending=request.args.get("validation_ordered"),
+        has_clinvar_submission=request.args.get("clinvar_submitted"),
+        projection=ALL_CASES_PROJECTION,
+    )
+    all_cases = _sort_cases(data, request, all_cases)
+
+    nr_cases = 0
+    for case_obj in all_cases:
```

Comment:
We could limit this to `limit + nr_cases_showall_statuses` - cases beyond that would never be used before the break below. But Im not clear on how much resources on the db that would save - the loop should not instantiate all cases here, right?